narwhals/dataframe.py

```diff
@@ -628,6 +628,42 @@ def columns(self) -> list[str]:
         """
         return super().columns
 
+    def rows(
+        self, *, named: bool = False
+    ) -> list[tuple[Any, ...]] | list[dict[str, Any]]:
```

Comment:
I think it would be a great addition to add type hinting for the two cases:

```suggestion     
    @overload
    def rows(
        self, *, named: Literal[True]
    ) -> list[dict[str, Any]]:

    @overload
    def rows(
        self, *, named: Literal[False]
    ) -> list[tuple[Any, ...]]:
    
    @overload
    def rows(
        self, *, named: bool = False
    ) -> list[tuple[Any, ...]] | list[dict[str, Any]]:
    
    def rows(
        self, *, named: bool = False
    ) -> list[tuple[Any, ...]] | list[dict[str, Any]]:
```

Reply:
Addressed.

Reply:
I was wondering if we want to put these overloads in an `if TYPE_CHECKING:` block. What do you think @FBruzzesi ?

Reply:
Thanks for addressing it!

> What do you think @FBruzzesi ?

This is personal taste grey zone! I will let @MarcoGorelli make the final call 😁